lib/updates/on_order.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const getPosition = require('../data/get_position')
+const tradeForOrder = require('../orders/trade_for_order')
+const calculateFees = require('../orders/calculate_fees')
+const updatePositionWithTrade = require('../data/update_position_with_trade')
+const debug = require('debug')('bfx:hf:strategy:updates:on-order')
+const updateStateOnClose = require('../position/update_state_on_close')
+const { STOP_ORDER_TYPE } = require('../constants')
+
+/**
+ * Called for a closed order
+ *
+ * @param {Object} state
+ * @param {Object} order
+ * @return {Promise} p - resolves to nextState
+ */
+const onOrder = async (state = {}, order = {}) => {
+  // check if stop order linked to the position is filled
+  if (order.type === STOP_ORDER_TYPE && order.amount === 0) {
```

Comment:
I suggest using "early return" to those if statements

Reply:
Fixed.